x-pack/plugins/timelines/public/components/utils/keury/index.test.ts

```diff
@@ -62,4 +62,28 @@ describe('Kuery escape', () => {
     const expected = 'This\\nhas\\tnewlines\\r\\nwith\\ttabs';
     expect(escapeKuery(value)).to.be(expected);
   });
+
+  it('should escape backslashes', () => {
+    const value = 'This\\has\\backslashes';
+    const expected = 'This\\\\has\\\\backslashes';
+    expect(escapeKuery(value)).to.be(expected);
+  });
+
+  it('should escape multiple backslashes and quotes', () => {
+    const value = 'This\\ has 2" quotes & \\ 2 "backslashes';
+    const expected = 'This\\\\ has 2\\" quotes & \\\\ 2 \\"backslashes';
+    expect(escapeKuery(value)).to.be(expected);
+  });
+
+  it('should escape all special character according to kuery.peg SpecialCharacter rule', () => {
+    /*
+     * Ref: packages/kbn-es-query/grammar/grammar.peggy
+     *
+     * SpecialCharacter
+     *  = [\\():<>"*{}]
+     */
+    const value = `\\():"*{}`;
+    const expected = `\\\\\\(\\)\\:\\"\\*\\{\\}`;
+    expect(escapeKuery(value)).to.be(expected);
```

Comment:
@XavierM , tagging you here as you may have move info about this.

Thare are couple of tests in this test suite which are failing because of my change. Do you think test below is correct. Even though test description says that all special characters should be replaced, cahracters such as `<` , `>` are not replaced. Is there a reson for that? 

https://github.com/elastic/kibana/blob/9fb17237953f9b1444d76436c8b20c9a1523ee36/x-pack/plugins/timelines/public/components/utils/keury/index.test.ts#L53-L59

Reply:
why not just using this package [packages/kbn-es-query/src/kuery/utils/escape_kuery.ts](https://github.com/elastic/kibana/blob/main/packages/kbn-es-query/src/kuery/utils/escape_kuery.ts) seems like it has been build by the people who know KQL. We did create this utility before because they did not exist but now they seem to be there for us. Simpler and focus on what we know! <3